pkg/nodebootstrap/userdata_al2.go

```diff
@@ -56,3 +61,11 @@ func NewUserDataForAmazonLinux2(spec *api.ClusterConfig, nodeGroupID int) (strin
 	logger.Debug("user-data = %s", body)
 	return body, nil
 }
+
+func makeKubeletParamsAmazon(spec *api.ClusterConfig, ng *api.NodeGroup) []string {
+	params := makeKubeletParamsCommon(spec, ng)
+	if len(ng.Labels) > 0 {
+		params = append(params, fmt.Sprintf("KUBELET_EXTRA_ARGS=--node-labels=%s", ng.Labels))
```

Comment:
I don't think `KUBELET_EXTRA_ARGS` will not work as we override the whole [kubelet `ExecStart` already](https://github.com/weaveworks/eksctl/blob/master/pkg/nodebootstrap/assets/10-eksclt.al2.conf).
We should just add the `NODE_LABELS` to `kubelet.env`, and add `--node-labels=${NODE_LABELS}` to `10-eksclt.al2.conf`.

By the way, we have switched to using config file on AL2 (#353), but it's not generated yet (#156) and it's not clear if Ubuntu supports this yet or not (#355). Once we have the config file fully generated, we will be able to get rid of the `kubelet.env`.

Reply:
My idea was `KUBELET_EXTRA_ARGS` works because I added it for both AL2 and Ubuntu in this PR, https://github.com/weaveworks/eksctl/pull/332/files/aee3198a4afda26ec4b433ceb3e3955758ac341b#diff-6955196fab7d3ecf801e80fe59c65f30R32 and in https://github.com/weaveworks/eksctl/pull/332/files/aee3198a4afda26ec4b433ceb3e3955758ac341b#diff-3efef590951f408a697d6c592491d9c4R43.

But `NODE_LABELS` would also be fine for clarity.

I agree we should migrate to KubeletConfiguration, and then to the kubelet componentconfig.